play-2.4/swagger-play2/app/play/modules/swagger/routes/RoutesFileParser.scala

```diff
@@ -0,0 +1,367 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package play.modules.swagger.routes
+
+import java.io.File
+
+
+import play.api.Logger
+import play.modules.swagger.SwaggerUtils
+import play.modules.swagger.routes.Route
+
+import scala.io.Codec
+import scala.language.postfixOps
+import scala.util.{Right, Left}
+import scala.util.parsing.combinator._
+import scala.util.parsing.input._
+
+
+/**
+  * backport form playframework compile
```

Comment:
I asked the Play dev team and they confirmed there's no way to introspect the routes at runtime to get this information and that your approach here is the only way to get the info at the moment